srcs/wordExpansion/get_variable_key.c

```diff
@@ -0,0 +1,91 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   get_variable_key.c                                 :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: brappo <brappo@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/04/04 08:34:50 by brappo            #+#    #+#             */
+/*   Updated: 2024/04/05 16:20:57 by brappo           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+bool	is_char_name(char c)
+{
+	if (ft_isalnum(c))
+		return (true);
+	if (c == '_')
+		return (true);
+	return (false);
+}
+
+ssize_t	find_bracket(char *input, ssize_t variable_start)
+{
+	char	*closing_bracket;
+
+	closing_bracket = ft_strchr(input + variable_start, '}');
+	if (closing_bracket == NULL)
+		return (-1);
+	return (closing_bracket - input);
+}
+
+bool	is_string_name(char *input, ssize_t start, ssize_t end)
+{
+	ssize_t	index;
+
+	index = start;
+	if (input == NULL || start < 0)
+		return (false);
+	while (input[index] && index < end)
+	{
+		if (!is_char_name(input[index]))
+			return (false);
+		index++;
+	}
+	return (true);
+}
+
+ssize_t	end_of_name(char *input, ssize_t variable_start)
```

Comment:
Penser à rendre statiques les fonctions qui n'ont pas besoin d'être visibles depuis l'extérieur du fichier.
Un [article](https://www.upgrad.com/tutorials/software-engineering/c-tutorial/static-function-in-c/) sur l'intérêt des fonctions statiques, en plus des conflits de noms.